src/mgmt/srv_pool.c

```diff
@@ -1009,3 +1009,38 @@ ds_mgmt_pool_delete_acl(uuid_t pool_uuid, const char *principal,
 out:
 	return rc;
 }
+
+int
+ds_mgmt_pool_set_prop(uuid_t pool_uuid, daos_prop_t *prop)
+{
+	int              rc;
+	d_rank_list_t   *ranks;
+	struct mgmt_svc	*svc;
+
+	if (prop == NULL) {
```

Comment:
This function could be used to set props with multiple entries though, right? So maybe assert  that `prop->dpp_br > 0` and `prop->dpp_entries` is non-NULL?

Reply:
How is this?
```c
	if (prop == NULL || prop->dpp_entries == NULL || prop->dpp_nr < 1) {
		D_ERROR("invalid property\n");
		rc = -DER_INVAL;
		goto out;
```

Reply:
Yes it would seem possible that this function could be used to set multiple properties at a time though I don't know if we have an API or tool that would expose the capability.

I think that logic looks reasonable yes.